src/main/service/listManipulation/crownFirmListService.ts

```diff
@@ -0,0 +1,114 @@
+                  caseSeparator: thisCase['caseSequenceIndicator'],
+                  linkedCases: thisCase['linkedCases'],
+                  hearingNotes: hearing['listingNotes'],
+                  defendant: formattedName[1].toUpperCase() + ', ' + formattedName[0],
+                  defendantRepresentative: hearing['defendantRepresentative'],
+                  prosecutingAuthority: hearing['prosecutingAuthority'],
+                  hearingType: hearing['hearingType'],
+                  hearingPlatform: sitting['caseHearingChannel'],
+                };
+                rows.push(row);
+              });
+            });
+          });
+        });
+      });
+    });
+    return this.splitByCourtAndDateAndAllocation(rows);
+  }
+
+  /**
+   * Splits the formatted list from above into three different dimensions:
+   * 1) First the cases are split by courtHouse
+   * 2) Then, the data within each courthouse is split by day
+   * 3) Then the data for each day/courtHouse pair is split by courtRoom.
+   * After the splitting, the method ensures that all unallocated cases always fall at the end of the cases in the
+   * current courtHouse/day/courtRoom using some array shuffling magic.
+   * @param data
+   * @private
+   */
+  private splitByCourtAndDateAndAllocation(data: any) {
+    const courts = [];
+    const uniqueCourts = dataManipulationService.uniquesInArrayByAttrib(data, 'courtName');
+    let courtCounter = 0;
+
+    uniqueCourts.forEach(court => {
+      const courtData = data.filter(row => row.courtName === court);
+      courts.push({'courtName': court, days: []});
+      const uniqueDays = dataManipulationService.uniquesInArrayByAttrib(courtData, 'sittingDate');
+      const uniqueDaysArr = [];
+      Array.from(uniqueDays).forEach(day => {
+        const encDay = moment.utc(day, 'dddd DD MMMM YYYY').tz(this.timeZone);
+        uniqueDaysArr.push(encDay);
+      });
+      uniqueDaysArr.sort(function(a, b) {
+        return a - b;
+      });
+      uniqueDaysArr.forEach(day => {
+        const thisDayCourts = [];
+        const formattedDay = moment.utc(day).tz(this.timeZone).format('dddd DD MMMM YYYY');
+        const record = courtData.filter(row => row.sittingDate === formattedDay);
+        const uniqueCourtRooms = dataManipulationService.uniquesInArrayByAttrib(record, 'courtRoom');
+        Array.from(uniqueCourtRooms).forEach(courtRoom => {
+          const room = record.filter(row => row.courtRoom === courtRoom);
+          thisDayCourts.push({'courtRoom': courtRoom, data: room});
+        });
+        // custom sort def - basically if it's got the string, move to end. Needed to be suppressed because eslint
+        // does not understand that a compare function needs two vars.
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        function compare (a, b) {
+          if (a.courtRoom.toLowerCase().includes('to be allocated')){return 1;} return -1;}
```

Comment:
The text "to be allocated" can come in both direction. You are comparing just one direction here. So any text which is not "to be allocated" will get bumped up as you return '-1'. So you are likely to have the order of the array reversed for all other texts. For example if you have an array ['a', 'b', 'to be allocated', 'c', 'd'], the result will be ['d', 'c', 'b', 'a', 'to be allocated']. 

It is fine if that is what you intend to do. If you want to enforce the order ['a', 'b', 'c', 'd', 'to be allocated'] might be better to consider both directions with a compare method like below:

function compare (a, b) {
      if (a.toLowerCase().includes('to be allocated')) {
        return 1;
      }
      if (b.toLowerCase().includes('to be allocated')) {
        return -1;
      }
      return 0;
    }

Reply:
Yeah, I was gonna do this at initial implementation but i just thought it wasn't really a requirement. Can add now.

Reply:
ok if we don't care about the ordering it probably doesn't matter but having the second comparison part will at least maintain the original order for the other elements.

Reply:
added btw